### PR TITLE
Add anatomy of a component to README

### PR DESCRIPTION
**Pull request by DTCurrie**
Starting to bring in things from https://github.com/viamrobotics/prime/pull/357 starting with this doc change. Feel free to leave edits as suggestions.

### CODE REVIEW

File: packages/core/README.md [hunk 1]
```diff
@@ -80,3 +80,84 @@ pnpm -C packages/core test:watch  # watch tests
...
+  $: {
+    someString = prop ? 'whoa' : 'no';
+  }
+
+  // lifecycle hooks
+  onMount(() => ...);
+</script>
+
+<!-- Your layout -->
+<div class="border-black"><slot /></div>
```
Comment (ethanlook) [inline packages/core/README.md:158]: Maybe something about using default slot when there's only one slot, but named slots when there are multiple?

**Review by ethanlook — APPROVED** [review]
💪🏼

File: packages/core/README.md (hunk 1 above)
Comment (DTCurrie) [reply packages/core/README.md:158]: Good callout

**Review by micheal-parks — APPROVED** [review]
🚀

**Pull request merged**